examples/server/server.cpp

```diff
@@ -950,18 +996,27 @@ int main(int argc, char ** argv) {
         res.status = 500;
     });
 
+    svr.set_error_handler([](const Request &, Response & res) {
+        res.set_content("File Not Found", "text/plain");
+        res.status = 404;
+    });
+
+
     // set timeouts and change hostname and port
     svr.set_read_timeout(sparams.read_timeout);
     svr.set_write_timeout(sparams.write_timeout);
 
     if (!svr.bind_to_port(sparams.hostname, sparams.port)) {
-        LOG_ERROR("couldn't bind to server socket", {
-            { "hostname", sparams.hostname },
-            { "port", sparams.port },
-        });
+        fprintf(stderr, "\ncouldn't bind to server socket: hostname=%s port=%d\n\n", sparams.hostname.c_str(), sparams.port);
         return 1;
     }
 
+    // Set the base directory for serving static files
+    svr.set_base_dir(sparams.public_path);
+
+    // to make it ctrl+clickable:
+    fprintf(stdout, "\nllama server listening at http://%s:%d\n\n", sparams.hostname.c_str(), sparams.port);
```

Comment:
```suggestion
    fprintf(stderr, "\nllama server listening at http://%s:%d\n\n", sparams.hostname.c_str(), sparams.port);
```

This needs to not go out with the JSON logs otherwise they could not be parsed correctly.

I think `stderr` is OK for this because other parts of llama.cpp also log there.